cpp/ql/src/semmle/code/cpp/internal/QualifiedName.qll

```diff
@@ -0,0 +1,343 @@
+class Namespace extends @namespace {
```

Comment:
This has the same name as a class in `Namespace.qll`, as do most of the other classes in this file.  Are we relying on (QL) namespaces to make them distinct?  Is it worth explaining what's going on in a comment?

Reply:
Yes, I’ll add a comment at the top of the file to explain what’s going on. 

The reason for writing this file directly on the dbscheme is that qualified names are needed to improve the ResolveClass mechanism, and ResolveClass sits below the AST classes. That’s what Ian was doing on the branch I took this from.